drivers/soundwire/intel.c

```diff
@@ -1544,15 +1545,20 @@ static int intel_master_remove(struct platform_device *pdev)
 	return 0;
 }
 
-int intel_master_process_wakeen_event(struct platform_device *pdev)
+static void intel_link_shutdown(struct ancillary_device *ancildev)
 {
-	struct device *dev = &pdev->dev;
+	intel_link_remove(ancildev);
+}
+
+int intel_link_process_wakeen_event(struct intel_link_dev *ldev)
+{
+	struct device *dev = &ldev->ancildev.dev;
 	struct sdw_intel *sdw;
 	struct sdw_bus *bus;
 	void __iomem *shim;
 	u16 wake_sts;
 
-	sdw = platform_get_drvdata(pdev);
+	sdw = dev_get_drvdata(dev);
```

Comment:
@plbossart this is confusing. in intel_link_probe() you call dev_set_drvdata() with cdns argument which is of type sdw_cdns but here you're trying to retreive a struct sdw_intel handle. Where is this overwritten?

Reply:
it's the same pointer in the end, we'll add a separate patch to clarify.